### PR TITLE
Remove Conversation from MII as it was deprecated and removed from transformers.

### PR DESCRIPTION
**Pull request by loadams**
Fixes in MII that correspond to [these changes](https://github.com/microsoft/DeepSpeed/pull/5629) in DeepSpeed

### CODE REVIEW

**Review by adk9 — APPROVED** [review]

**Comment by mrwyattii** [comment]
Need to remove the conversational task type unit test, set minimum version of transformers to whatever version removed the task type, and update docs/READMEs to reflect these changes

File: requirements/requirements.txt [hunk 1]
```diff
@@ -10,6 +10,6 @@ pydantic
 pyzmq
 safetensors
 torch
-transformers
+transformers>=4.41.2
```
Comment (adk9) [inline requirements/requirements.txt:13]: `ConversationalPipeline` and `Conversation` has been deprecated for a while and was removed in transformers 4.41.2. Do we really have to pin this version as minimum? These changes should be backward-compatible.

File: mii/legacy/examples/local/conversational-example.py [hunk 2]
```diff
@@ -11,4 +11,4 @@
 
 print(f"Deploying {name}...")
 
-mii.deploy(task='conversational', model=name, deployment_name=name + "_deployment")
+mii.deploy(task='question-answering', model=name, deployment_name=name + "_deployment")
```
Comment (adk9) [inline mii/legacy/examples/local/conversational-example.py:14]: `TextGenerationPipeline` is recommended as replacement for `ConversationalPipeline`, so you likely want `task='text-generation'` here.

File: requirements/requirements.txt (hunk 1 above)
Comment (loadams) [reply requirements/requirements.txt]: Good point, unpinning.

File: mii/legacy/examples/local/conversational-example.py (hunk 2 above)
Comment (loadams) [reply mii/legacy/examples/local/conversational-example.py]: Good catch, updated.
